src/NHibernate.Test/Async/Linq/LinqQuerySamples.cs

```diff
@@ -83,56 +83,35 @@ public class LinqQuerySamplesAsync : LinqTestCase
 							" name and product availability.")]
 		public Task DLinq13Async()
 		{
-			try
-			{
-				var q =
+			var q =
 				from p in db.Products
 				select new {p.Name, Availability = p.UnitsInStock - p.UnitsOnOrder < 0 ? "Out Of Stock" : "In Stock"};
 
-				return ObjectDumper.WriteAsync(q, 1);
-			}
-			catch (Exception ex)
-			{
-				return Task.FromException<object>(ex);
-			}
+			return ObjectDumper.WriteAsync(q, 1);
```

Comment:
@maca88 is there a way to turn off pass-through optimization for async code? Eg. always generate async-await?

Reply:
Currently there isn't. Do you want an `async-await` in order to have this line in the stack trace or is something else that you want to achieve?

Reply:
The intention is to get clearer code for tests, or to be able to opt-out

Reply:
[Will add an option](https://github.com/maca88/AsyncGenerator/issues/97) for this.